widgets/src/components/SummaryCard/SummaryCardConfig.js

```diff
@@ -0,0 +1,151 @@
+      settings: '{"summaryId": ""}',
+      knowledgeSource: '',
+    };
+
+    this.onChangeKnowledgeSource = this.onChangeKnowledgeSource.bind(this);
+    this.onChangeSettings = this.onChangeSettings.bind(this);
+    this.handleSave = this.handleSave.bind(this);
+  }
+
+  async componentDidMount() {
+    const { frameId, pageCode } = this.props;
+
+    // getting list of Kie server connections
+    const sourceList = await getConnections();
+    this.setState({ sourceList: sourceList.payload });
+
+    // getting existing configs
+    const pageWidgetsConfigs = await getPageWidget(pageCode, frameId, 'SUMMARY_CARD');
+
+    const configs = pageWidgetsConfigs.payload && pageWidgetsConfigs.payload.config;
+
+    if (configs && configs.knowledgeSource) {
+      this.onChangeKnowledgeSource(configs.knowledgeSource, () => {
+        if (configs.settings) {
+          this.setState({
+            settings: configs.settings,
+          });
+        }
+      });
+    }
+  }
+
+  onChangeKnowledgeSource(e, cb = () => {}) {
+    const knowledgeSource = e.target ? e.target.value : e;
+    this.setState({ knowledgeSource }, cb);
+  }
+
+  onChangeSettings({ target: { value } }) {
+    this.setState({ settings: value });
+  }
+
+  async handleSave() {
+    const { frameId, pageCode, widgetCode } = this.props;
+    const { knowledgeSource, settings } = this.state;
+
+    const body = JSON.stringify({
+      code: widgetCode,
+      config: {
+        knowledgeSource,
+        settings,
+      },
+    });
+
+    try {
+      const response = await putPageWidget(pageCode, frameId, body);
+      console.log('Configs got saved', response);
+    } catch (error) {
+      console.log('Error while saving configs', error);
+    }
+  }
+
+  render() {
+    const { knowledgeSource, sourceList, settings } = this.state;
+
+    return (
+      <div>
+        <form>
+          <Row>
+            <Col xs={12}>
+              <FormGroup controlId="connection">
+                <ControlLabel>Knowledge Source</ControlLabel>
+                <select
+                  className="form-control"
+                  value={knowledgeSource}
+                  onChange={this.onChangeKnowledgeSource}
+                >
+                  <option value="">Select...</option>
+                  {sourceList.map(source => (
+                    <option key={source.name} value={source.name}>
+                      {source.name}
+                    </option>
+                  ))}
+                </select>
+                <HelpBlock>Select one of the Kie server connections.</HelpBlock>
+              </FormGroup>
+            </Col>
+          </Row>
+          {knowledgeSource && (
+            <section>
+              <legend>Settings</legend>
+              <Row>
+                <Col xs={12}>
+                  <FormGroup bsClass="form-group" controlId="textarea">
+                    <ControlLabel bsClass="control-label">Config</ControlLabel>
+                    <FormControl
```

Comment:
that should be a Select with a list of `summaries`, allowing users to pick only the existing ones.

it should be fetched from `getSummaries` method

Reply:
done. select field for summaries has been added